packages/synapse-react-client/src/components/CannedRejectionDialog/CannedRejectionDialog.tsx

```diff
@@ -0,0 +1,400 @@
+import {
```

Comment:
File for a new component that renders the dialog that can be used to craft a rejection message.

Very little new logic/changes (called out). Mostly just refactored.